packages/@sanity/form-builder/src/inputs/Array/ItemValue.js

```diff
@@ -161,11 +175,42 @@ export default class RenderItemValue extends React.Component<Props> {
       )
     }
 
+    if (options.editModal === 'popover') {
+      return (
+        <div className={styles.popupAnchor}>
+          <EditItemPopOver onClose={this.handleEditStop} key={item._key}>
+            {content}
+          </EditItemPopOver>
+        </div>
+      )
+    }
+
     return (
       <div className={styles.popupAnchor}>
-        <EditItemPopOver onClose={this.handleEditStop} key={item._key}>
-          {content}
-        </EditItemPopOver>
+        <DefaultDialog
+          isOpen
```

Comment:
Is `isOpen` this really needed? IMO if the dialog shouldn't be open, it should not be rendered at all.

Reply:
Now it is needed, but I agree. Lets do that in another branch since it potentially requires a lot of changes

Reply:
Agreed!